lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HadoopFileSystem.cs

```diff
@@ -118,46 +118,108 @@ public Stream Create(Uri fileUri)
         public void Delete(Uri fileUri)
         {
             // Delete the file via the hdfs command line.
-            _commandRunner.Run("dfs -rm " + fileUri.AbsolutePath);
+            try
+            {
+                _commandRunner.Run("dfs -rm " + fileUri);
+            }
+            catch
```

Comment:
This assumes that the command runner throws an exception. Is there a way to validate that assumption?

Reply:
I don't think it throws an exception for a bad URI.

[This line](https://github.com/apache/reef/blob/85bae60ab0b988b143ddd49f8f4295475ebdf49c/lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HDFSCommandRunner.cs#L133) is where `Run()` throws an exception, and it's if the underlying call timed out `_numberOfRetries` times.

In the case of a missing file, we get a valid `CommandResult` object, but it probably specifies that no file was found (all I know is that no result is thrown). I'm guessing the same is true for the case of a bad server name. @shouhengyi-microsoft  What do you see on your cluster with this behavior? Does this branch work?